tests/functional/ingestion/IngestionReader.js

```diff
@@ -103,112 +102,83 @@ describe('ingestion reader tests with mock', function fD() {
     let httpServer;
     let producer;
     let zkClient;
-
-    before(done => {
+    const mongoUrl =
+    `mongodb://${testConfig.queuePopulator.mongo.replicaSetHosts}` +
+    '/db?replicaSet=rs0';
+    const client = new MongoClient(mongoUrl, {});
+    let db;
+    before(async () => {
         testConfig.s3.port = testPort;
-        const mongoUrl =
-            `mongodb://${testConfig.queuePopulator.mongo.replicaSetHosts}` +
-            '/db?replicaSet=rs0';
         const topic = testConfig.extensions.ingestion.topic;
-        async.waterfall([
-            next => {
-                MongoClient.connect(mongoUrl, {}, (err, client) => {
-                    if (err) {
-                        next(err);
-                    }
-                    this.client = client;
-                    this.db = this.client.db('metadata', {
-                        ignoreUndefined: true,
-                    });
-                    next();
-                });
-            },
-            next => kafkaAdminClient.createTopic({
+        await client.connect();
+        db = client.db('metadata', { ignoreUndefined: true });
+        try {
+            await new Promise((resolve, reject) => {
```

Comment:
may use util.promisify, and move error handlign to the try/catch block:

```suggestion
try {
   const createTopic = util.promisify(kafkaAdminClient.createTopic).bind(kafkaAdminClient);
   await createTopic({topic, num_partitions: 1, replication_factor: 1}) // eslint-disable-line camelcase
} catch(err) {
   if err.code !== 36) {
      throw err;
   }
}
```